libs/damap/src/lib/components/dmp/people/people.component.ts

```diff
@@ -43,21 +47,41 @@ export class PeopleComponent implements OnInit {
 
   constructor(
     private backendService: BackendService,
-    public dialog: MatDialog,
-  ) {}
+    public dialog: MatDialog
+  ) { }
 
   ngOnInit(): void {
-    this.backendService.loadServiceConfig().subscribe(service => {
-      this.serviceConfig$ = service.personSearchServiceConfigs;
-      this.serviceConfigType = this.serviceConfig$[0];
-    });
-    this.searchResult$ = this.searchTerms.pipe(
-      debounceTime(300),
-      distinctUntilChanged(),
-      switchMap((term: string) =>
-        this.backendService.getPersonSearchResult(term, this.serviceConfigType.displayText)
+    this.backendService
+      .loadServiceConfig()
+      .pipe(
+        tap(serviceConfigType => {
+          this.serviceConfig$ = serviceConfigType.personSearchServiceConfigs;
+          this.serviceConfigType = this.serviceConfig$[0];
+        }),
+        // operator is used to switch to a new Observable stream
+        switchMap(() =>
+          this.searchTerms.pipe(
+            debounceTime(300),
+            distinctUntilChanged(),
+            switchMap(term =>
+              this.backendService.getPersonSearchResult(
+                term,
+                this.serviceConfigType.displayText
+              )
+            )
+          )
+        )
       )
-    );
+      //  method is used to subscribe to the search result stream and update 
+      //  the searchResult$ within the latest search results
+      .subscribe(result => {
+        this.searchResult$ = of(result);
+      });
+  }
```

Comment:
major: this seems rather complicated and there are many things happening at once. How about subscribing to the terms pipe only for fetching the results?
e.g.
```
// fetch config
    this.backendService.loadServiceConfig().subscribe(service => {
      this.serviceConfig$ = service.personSearchServiceConfigs;
      this.serviceConfigType = this.serviceConfig$[0];
    });

// subscribe to terms and perform people search
    this.searchTerms
      .pipe(debounceTime(300), distinctUntilChanged())
      .subscribe((term: string) => {
        this.searchResult$ = this.backendService.getPersonSearchResult(
          term,
          this.serviceConfigType.displayText
        );
      });
```

Either way, we should probably store the subscriptions and cancel them in `onDestroy`